File: level2/ex28.js
// 진법 변환 함수
function change_base(n, base=3, base_char='412') {
  const mod = n % 3;
  const div = mod % 3 === 0 ? parseInt(n / 3) - 1 : parseInt(n / 3);
  
  if (div === 0) return base_char[mod];
  else return change_base(div) + base_char[mod]
}

function solution(n) {
  return change_base(n);
}

File: level2/ex28.py
# 진법 변환 함수
def change_base(n, base=3, base_char='412'):
    div, mod = divmod(n, base)

    if mod % 3 == 0:
        div -= 1
    
    if div == 0:
        return base_char[mod]
    
    return change_base(div) + base_char[mod]

def solution(n):
    return change_base(n)